template/src/pages/Demo/SubDemo/SubDemo.tsx

```diff
@@ -0,0 +1,13 @@
+import { useOutletContext, useParams } from 'react-router';
+
+export const SubDemo = () => {
+  const { id = '' } = useParams<{ id: string }>();
+  const context = useOutletContext<string>();
+
```

Comment:
Cela permet une meilleure gestion non ?
```suggestion
  const { id } = useParams<{ id: string }>();
  const context = useOutletContext<string>();

  if (!id) {
    return null;
  }
```

Reply:
alors ici le cas ou l'id est `undined` n'arrivera jamais du coup est ce vraiment utile d'ajouter une branche de code qu'il faudra tester et qui sera littéralement du code mort ?

Reply:
même dans le cas où un utilisateur tape l'url à la main ?

Reply:
oui si tu met `/demo/` tu arrive sur la page demo mais la sous page demo ne s'affiche pas